testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/sourceProvider/SourceProviderTest.java

```diff
@@ -0,0 +1,80 @@
+package org.jboss.resteasy.test.sourceProvider;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.test.sourceProvider.resource.Book;
+import org.jboss.resteasy.test.sourceProvider.resource.BookResource;
+import org.jboss.resteasy.test.sourceProvider.resource.SourceProviderApp;
+import org.jboss.resteasy.util.HttpResponseCodes;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.*;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Response;
+import javax.xml.transform.stream.StreamSource;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class SourceProviderTest {
+
+    private static Client client;
+    private static final String PATH = "http://localhost:8080/sourceProvider/test";
```

Comment:
This is not ok. Please have a look at other tests and see how resource addresses are built. E.g. https://github.com/resteasy/Resteasy/blob/master/testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/interceptor/StreamCloseTest.java#L55